osutil/epoll/epoll.go

```diff
@@ -54,27 +78,36 @@ func Open() (*Epoll, error) {
 	e := &Epoll{
 		fd:                fd,
 		registeredFdCount: 0,
+		closed:            make(chan struct{}),
 	}
 	runtime.SetFinalizer(e, func(e *Epoll) {
```

Comment:
I believe ensures that the epoll FD gets closed properly if the epoll instance gets garbage collected without being closed by the user. Otherwise, the kernel would sit there collecting r/w events to send over a FD which will never be read.